transforms/language/doc_quality/python/src/doc_quality_transform.py

```diff
@@ -0,0 +1,233 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+from argparse import ArgumentParser, Namespace
+from typing import Any
+
+import os
+import pyarrow as pa
+from data_processing.data_access import DataAccessFactory, DataAccess
+from data_processing.transform import AbstractTableTransform, TransformConfiguration
+from data_processing.utils import CLIArgumentProvider, TransformUtils, get_logger
+from doc_c4_statistics import (
+    c4_contain_pattern_ratio,
+    c4_contains_ldnoobw_words,
+    c4_load_ldnoobw_words,
+    c4_sentence_count,
+)
+from doc_Gopher_statistics import (
+    compute_average_japanese_sentence_length,
+    compute_bullet_point_ellipsis_alphabet_word_ratio,
+    compute_word_statistics,
+    contains_common_English_words,
+    find_first_japanese_alphabet_position,
+)
+
+logger = get_logger(__name__)
+
+short_name = "docq"
+cli_prefix = f"{short_name}_"
+text_lang_key = "text_lang"
+doc_content_column_key = "doc_content_column"
+bad_word_filepath_key = "bad_word_filepath"
+text_lang_cli_param = f"{cli_prefix}{text_lang_key}"
+doc_content_column_cli_param = f"{cli_prefix}{doc_content_column_key}"
+bad_word_filepath_cli_param = f"{cli_prefix}{bad_word_filepath_key}"
+
+default_text_lang = "en"
+default_doc_content_column = "contents"
+
+data_factory_internal_key = f"{cli_prefix}data_factory"
+files_to_use_internal_key = f"{cli_prefix}files_to_use"
+
+class DocQualityTransform(AbstractTableTransform):
+    """
+    Implements a transform to calculate document quality.
+    """
+
+    def __init__(self, config: dict[str, Any]):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments defined
+        by the companion runtime, DocQualityTransformRuntime.
+        """
+        # Make sure that the param name corresponds to the name used in apply_input_params method
+        # of DocQualityTransformConfiguration class
+        super().__init__(config)
+        self.text_lang = config.get(text_lang_key, default_text_lang)
+        self.doc_content_column = config.get(doc_content_column_key, default_doc_content_column)
+        
+        daf = config.get(data_factory_internal_key, None)
+        bad_word_filepath = config.get(bad_word_filepath_key, None)
+        if bad_word_filepath is None:
+            raise RuntimeError(f"Missing configuration value for key {bad_word_filepath_key}")
+        if os.path.exists(bad_word_filepath):
+            logger.info(f"Load badwords found locally from {bad_word_filepath}")
+            self.re_pattern = c4_load_ldnoobw_words(ft_lang=self.text_lang, file_path=bad_word_filepath)
+        else:
+            logger.info(f"Load badwords from remote")
+            data_access = daf.create_data_access()
```

Comment:
Should throw an exception if daf == None.

Reply:
Done

Reply:
Suggest the following `"Did not find DataAccessFactory instance under f{data_factory_internal} key.  This is required when bad word file is not in the local file system"`